### ISSUE TITLE
Add current workdir to pythonpath automatically

### ISSUE THREAD

**Issue by akelad** [RasaHQ/rasa#3883]
**Description of Problem**:
Using the new `rasa` commands doesn't automatically load the current workdir as a potential module directory anymore, which means peoples custom components don't work (https://github.com/RasaHQ/rasa/issues/3748). They have to manually add them to their PYTHONPATH, which with the old `python -m` way wasn't necessary

**Overview of the Solution**:
We should consider adding the current workdir as a potential module directory automatically on startup

**Comment by erohmensing** [comment]
Oops I assigned before noticing the discussion label. Also I noticed it's enhancement, but something doesn't work that did before -- I'd call that a bug.

We already handle this for `rasa run actions` here: https://github.com/RasaHQ/rasa/blob/master/rasa/cli/run.py#L52

I've tried the same thing for `rasa train` and it works. We'd need to do the same with most of the other cli commands (`shell`, `run`, `test`, `interactive`). What's the discussion here?

**Comment by akelad** [comment]
https://github.com/RasaHQ/rasa/issues/3867

**Comment by akelad** [comment]
wait i'm confused are we even discussing this anymore or just doing it ? :D @wochinge I think you suggested we discuss it right? Any arguments against doing this?

**Comment by wochinge** [comment]
@erohmensing Unless something is exploding, we should take things only from the backlog. 

I think that's the most dangerous thing which I could think off what could happen
http://python-notes.curiousefficiency.org/en/latest/python_concepts/import_traps.html#the-name-shadowing-trap

Ellas implementation could also lead to very weird behavior if you are running rasa in the cloned `rasa` repository (http://python-notes.curiousefficiency.org/en/latest/python_concepts/import_traps.html#the-double-import-trap) 

I think we can do it, if we are getting complaints / weird bugs we can remove it again.

**Comment by akelad** [comment]
yeah i already told ella that :D

**Comment by erohmensing** [comment]
> Ellas implementation could also lead to very weird behavior if you are running rasa in the cloned rasa repository

We all work in a cloned repo for development, do we think it'll affect us there?

> I think we can do it, if we are getting complaints / weird bugs we can remove it again.

If not, I say we throw this into my patch and see if we get complaints

**Comment by wochinge** [comment]
> We all work in a cloned repo for development, do we think it'll affect us there?

Well, I usually dont run Rasa in the cloned repo, so should be fine (don't know though how others use it)

> If not, I say we throw this into my patch and see if we get complaints

:+1:

**Comment by erohmensing** [comment]
I just mean when i'm testing commands and stuff. I'll add it to mine while I work on a PR today that involves testing commands and see if anything funky happens

**Comment by rgstephens** [comment]
@wochinge I'm trying to get my custom tracker working again. Haven't touched it since June and I'm still running into the issue described in [#3867](https://github.com/RasaHQ/rasa/issues/3867). I'm running under 1.3.9.  

Normally, I'd map the volume as I do with my actions file, but this is not working:

```yml
x-rasa-services: &default-rasa-service
  restart: always
  image: "rasa/rasa:${RASA_VERSION}-full"
  volumes:
  - ./trackerArango:/app/trackerArango
```

I've tried setting `PYTHONPATH` in the docker-compose but this also does not work.

```yml
x-rasa-services: &default-rasa-service
  restart: always
  image: "rasa/rasa:${RASA_VERSION}-full"
  environment:
  - PYTHONPATH=/app
```

**Comment by wochinge** [comment]
@rgstephens What about setting `PYTHONPATH=/app/trackerArango`?

**Comment by erohmensing** [comment]
@rgstephens based on the file structure in the container, I think you might be missing a step in how you define it: `trackerArango.ArangoTrackerStore` 

It should be the path to the class in the module (from the working directory). So I think it should be `trackerArango.trackerArango.ArangoTrackerStore`, right? 

Looking at 
```
I have no name!@32dd25bf7990:/app$ find .
.
./trackerArango
./trackerArango/trackerArango.py
./trackerArango/__init__.py
```

**Comment by rgstephens** [comment]
Thanks, @erohmensing but that didn't work.

```
Store type 'trackerArango.trackerArango.ArangoTrackerStore' not found. Using InMemoryTrackerStore instead
```

The customer is happy about the switch from Mongo to Postgres so I think we'll forget Arango. Let me know if you have other ideas for me to try but I'm moving on to other issues.

**Comment by surajpadhy** [comment]
Any luck with this issue?
Even i am getting it while trying to run the core server with:

rasa run --endpoints endpoints.yml --credentials credentials.yml -p 5002 --enable-api

Unable to load Custom Tracker.
2019-11-08 17:03:14 WARNING  rasa.core.tracker_store  - Store type 'utilities.tracker_store.MongoTracker' not found. Using InMemoryTrackerStore instead

**Rasa** : 1.4.3
**Python** : 3.6.8

### config.yml:

language: en
pipeline:
  - name: "WhitespaceTokenizer"
    intent_split_symbol: "-"
  - name: "RegexFeaturizer"
  - name: "EntitySynonymMapper"
  - name: "CRFEntityExtractor"
  - name: "CountVectorsFeaturizer"
  - name: "EmbeddingIntentClassifier"
    epochs: 45

policies:
  - name: KerasPolicy
    max_history: 2
    epochs: 75
    batch_size: 30
    augmentation: 0
  - name: AugmentedMemoizationPolicy
    max_history: 1
  - name: FormPolicy
  - name: MappingPolicy
  - name: TwoStageFallbackPolicy
    nlu_threshold: 0.7
    core_threshold: 0.3
    fallback_core_action_name: "action_default_ask_affirmation"
    fallback_nlu_action_name: "action_default_ask_affirmation"
    deny_suggestion_intent_name: "out_of_scope

**Comment by erohmensing** [comment]
@surajpadhy when loading a custom module, the relevant info would be your project directory and how you reference the module (i.e. here the endpoints.yml, since it's a tracker store). Can you provide those?

**Comment by surajpadhy** [comment]
@erohmensing 
Created a custom tracker called _MongoTracker_ in a file named _tracker_store.py_, stored inside utilities folder in the root of the project.
here is my custom tracker that i included in the endpoints.yml

tracker_store:
   type: utilities.tracker_store.MongoTracker
   url: mongodb://localhost:27017
   db: test
   username: ""
   password: ""

**Comment by shubh802** [comment]
@erohmensing 
Rasa 1.10.0. I am facing the same issue.
UserWarning: Tracker store with type 'custom_tracker.CustomTrackerStore' not found. Using `InMemoryTrackerStore` instead.
```
tracker_store:
  type: custom_tracker.CustomTrackerStore
  url: mongodb://localhost:27017
  db: "rasa"
  username: ""
  password: ""
```
```
class CustomTrackerStore(TrackerStore):
    def __init__(self,
                 domain,
                 url,
                 host="mongodb://localhost:27017",
                 db="rasa",
                 username=None,
                 password=None,
                 auth_source="admin",
                 collection="rasa_conversations", event_broker=None):
        self.client = MongoClient(
            host, username=username, password=password, authSource=auth_source, connect=False)
        self.db = Database(self.client, db)
        self.collection = collection
        super(CustomTrackerStore, self).__init__(domain, event_broker)

        self._ensure_indices()
```
Any help over here.

**Comment by wochinge** [comment]
@shubh802 Could you please ask the question in the [forum](https://forum.rasa.com/)? Both, Rasa team members and the Rasa community, will help you there 🙂 Thanks 🙏

**Comment by hozayfarifai** [comment]
I am using rasa version 1.10.7 with python 3.7 and I am having the same issue 

```
tracker_store:
   type: myTracker.MongoTrackerStore # mongod
   url: mongodb://localhost:27017
   db: nlu
```

`/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/rasa/utils/common.py:363: UserWarning: Tracker store with type 'myTracker.MongoTrackerStore' not found. Using `InMemoryTrackerStore` instead.`

**Comment by zhangjustin** [comment]
Seems i get same issue with rasa2.3.4
/opt/venv/lib/python3.8/site-packages/rasa/shared/utils/io.py:93: UserWarning: Tracker store with type 'cluster_tracker_store.RedisClusterTrackerStore' not found. Using `InMemoryTrackerStore` instead.

Any help over here?

**Comment by sunnyjocker** [comment]
> Seems i get same issue with rasa2.3.4
> 
> ```
> /opt/venv/lib/python3.8/site-packages/rasa/shared/utils/io.py:93: UserWarning: Tracker store with type 'cluster_tracker_store.RedisClusterTrackerStore' not found. Using `InMemoryTrackerStore` instead.
> ```
> 
> Any help over here?

any idea how to get this solved, rasa 2.8.26 still not work
